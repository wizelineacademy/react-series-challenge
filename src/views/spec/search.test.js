import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';

import { Search } from '../';

const mockStore = configureMockStore();

describe('<Search />', () => {
  const props = {
    fetchGifs: jest.fn(),
    match: { params: { text: 'wizeline' } },
  };

  test('renders null if no data and not loading', () => {
    const initialState = {
      favorites: [],
      search: {
        data: undefined,
        loading: false,
      },
    };
    const tree = renderer.create(
      <Provider store={mockStore(initialState)}>
        <StaticRouter location="" context={{}}>
          <Search {...props} />
        </StaticRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders loader', () => {
    const initialState = {
      favorites: [],
      search: {
        data: undefined,
        loading: true,
      },
    };
    const tree = renderer.create(
      <Provider store={mockStore(initialState)}>
        <StaticRouter location="" context={{}}>
          <Search {...props} />
        </StaticRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders error message', () => {
    const initialState = {
      favorites: [],
      search: {
        data: undefined,
        loading: true,
        error: { message: 'Something went wrong' },
      },
    };
    const tree = renderer.create(
      <Provider store={mockStore(initialState)}>
        <StaticRouter location="" context={{}}>
          <Search {...props} />
        </StaticRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders data correctly', () => {
    const initialState = {
      favorites: [],
      search: {
        data: {
          pagination: { count: 1, offset: 0, total_count: 1 },
          data: [
            {
              id: 'asdfljlk3-ffgfg',
              title: 'Example image',
              images: {
                fixed_width_downsampled: {
                  url: 'http://www.example.com/image.jpg',
                  width: '800',
                  height: '900',
                },
              },
              user: {
                username: 'krilbert',
                display_name: 'Alfer',
                profile_url: 'http://www.example.com',
                avatar_url: 'http://www.example.com/avatar.jpg',
              },
            },
          ],
        },
        loading: false,
      },
    };
    const tree = renderer.create(
      <Provider store={mockStore(initialState)}>
        <StaticRouter location="" context={{}}>
          <Search {...props} />
        </StaticRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
