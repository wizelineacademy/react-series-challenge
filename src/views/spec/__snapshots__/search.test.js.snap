// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders data correctly 1`] = `
<div>
  wizeline
  <div>
    Showing 1 GIFs of 1
    <div
      className="sc-bdVaJa kEWmmQ"
    >
      <section
        className="sc-bxivhb jDMTmg"
      >
        <a
          href="/gifs/asdfljlk3-ffgfg"
          onClick={[Function]}
        >
          <img
            alt="Example image"
            height={120}
            src="http://www.example.com/image.jpg"
            width={200}
          />
        </a>
        <div
          className="sc-ifAKCX drTRiD"
        >
          <a
            alt="Fav"
            className="sc-EHOje laSDVF"
            onClick={[Function]}
          >
            <i
              className="far fa-thumbs-up"
            />
          </a>
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`<Search /> renders error message 1`] = `
<div>
  Something went wrong
</div>
`;

exports[`<Search /> renders loader 1`] = `
<div>
  wizeline
  <div
    className="sc-bwzfXH kmXviU"
  >
    <div
      animation-delay={0}
      className="sc-htpNat gLLNR"
    />
    <div
      animation-delay="-1.1s"
      className="sc-htpNat gteQUn"
    />
    <div
      animation-delay="-1.0s"
      className="sc-htpNat dnZQtX"
    />
    <div
      animation-delay="-0.9s"
      className="sc-htpNat fqexcQ"
    />
    <div
      animation-delay="-0.8s"
      className="sc-htpNat gcoKLq"
    />
  </div>
</div>
`;

exports[`<Search /> renders null if no data and not loading 1`] = `null`;
